src/Message/MessageParser.ts

```diff
@@ -82,14 +84,15 @@ const tagUnescapeMap: { [char: string]: string } = {
 	s: ' '
 };
 
-export function parseTags(raw: string): Map<string, string> {
+export function parseTags(raw: string): Map<string, string> | undefined {
 	const tags: Map<string, string> = new Map();
 	const tagStrings = raw.split(';');
 	for (const tagString of tagStrings) {
 		const [tagName, tagValue] = splitWithLimit(tagString, '=', 2);
 		// unescape according to http://ircv3.net/specs/core/message-tags-3.2.html#escaping-values
+		if (tagName === '') continue; // Ignore empty tags: @ @; @x; etc.
 		tags.set(tagName, tagValue.replace(/\\([\\:nrs])/g, (_, match) => tagUnescapeMap[match]));
 	}
 
-	return tags;
+	return tags.size > 0 ? tags : undefined;
```

Comment:
This introduces unnecessary `undefined` checks every time you want to check a specific tag, i.e.
```js
msg.tags.get('foo')
```
would become
```js
msg.tags ? msg.tags.get('foo') : undefined
```

(Also a very unnecessary breaking change.)

Reply:
I don't believe that to be the case,
This will only return `undefined` if a valid tag does not exist; The same as it would if it did not find a tag. I checked this one out to make sure I didn't break anything.

As a side note, doesn't Map.get() return `V | undefined` anyway?

Reply:
`Map.prototype.get` is not the concern here. The problem is that you need to check whether the map is actually there.

I just checked, and the map can actually be undefined as a Message member (the getter remedies that), so you're right that this doesn't break anything, but I'd actually like to go the opposite direction (i.e. ensuring that the member is always set).

Reply:
As a bonus to this PR, Tags is always defined now.